frontend/src/containers/Collection/CollectionManage/CollectionManage.js

```diff
@@ -0,0 +1,167 @@
+import React, { Component } from "react";
+import { connect } from "react-redux";
+import PropTypes from "prop-types";
+import { Link } from "react-router-dom";
+import { Button } from "react-bootstrap";
+
+import { collectionActions } from "../../../store/actions";
+import { collectionStatus } from "../../../constants/constants";
+import { TransferOwnershipModal, WarningModal, ManageCollectionMemberModal } from "../../../components";
+
+import "./CollectionManage.css";
+
+class CollectionManage extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            collectionName: "",
+            collectionDescription: "",
+        };
+    }
+
+    componentDidMount() {
+        const collectionId = this.props.location.pathname.split("=")[1].split("/")[0];
+        this.props.onGetCollection({ id: collectionId })
+            .then(() => {
+                if (this.props.collectionStatus !== collectionStatus.SUCCESS) {
+                    this.props.history.push("/main");
+                } else {
+                    this.setState({
+                        collectionName: JSON.parse(JSON.stringify(
```

Comment:
Why does this code need JSON parse and stringify?

Reply:
You can just do like this.

```
                    this.setState({
                        collectionName: this.props.selectedCollection.title,
                        collectionDescription: this.props.selectedCollection.text,
                    });
```

Reply:
At first I had a trouble with passing props to state and make state changes (when passing props, those string hadn't been able to change at that time), so I used JSON parse things. But somehow  it works fine without them, so I deleted them in recent push.